docarray/helper.py

```diff
@@ -256,3 +259,52 @@ def _shallow_copy_doc(doc):
         setattr(shallow_copy, field_name, val)
 
     return shallow_copy
+
+
+def _is_tensor(x: Any) -> bool:
+    """
+    Determines whether `x` is either np.ndarray, torch.tensor, or tf.tensor
+    """
+
+    if is_torch_available():
+        import torch
+
+        if isinstance(x, torch.Tensor):
+            return True
+
+    if is_tf_available():
+        import tensorflow as tf  # type: ignore
+
+        t = getattr(x, 'tensor', None)
+        if tf.is_tensor(t):
+            return True
+
+    if isinstance(x, np.ndarray):
+        return True
+
+    return False
+
+
+def _tensor_equals(tens1: Any, tens2: Any) -> bool:
+    """
+    Determines if two {torch, tf, np} tensors are equal.
+    If at least one of them is not a tensor, of if they are tensors of different frameworks, False is returned.
+    """
+    if is_torch_available():
+        import torch
+
+        if isinstance(tens1, torch.Tensor) and isinstance(tens2, torch.Tensor):
+            return torch.equal(tens1, tens2)
+
+    if is_tf_available():
+        import tensorflow as tf  # type: ignore
+
+        t1, t2 = getattr(tens1, 'tensor', None), getattr(tens2, 'tensor', None)
+        if tf.is_tensor(t1) and tf.is_tensor(t2):
+            return t1.shape == t2.shape and tf.math.reduce_all(tf.equal(t1, t1))  # type: ignore
+
+    are_np_arrays = isinstance(tens1, np.ndarray) and isinstance(tens2, np.ndarray)
+    if are_np_arrays:
+        return np.array_equal(tens1, tens2)
```

Comment:
should not this be part of our computational backend ?

Reply:
good point, I will refactor

Reply:
thx